src/plugin.py

```diff
@@ -230,6 +229,18 @@ def create_total_games_cache(self):
         return cache
 
     if ARE_ACHIEVEMENTS_IMPLEMENTED:
+        async def prepare_achievements_context(self, game_ids: List[str]) -> Any:
+            achievements_cache = {}
+            for game_id in game_ids:
+                title_id = get_game_title_id_from_ros_title_id(game_id)
+                if not games_cache[title_id]["achievementId"]:
+                    continue
+                achievement_id = get_achievement_id_from_ros_title_id(game_id)
+                url = f"https://scapi.rockstargames.com/achievements/all?title={achievement_id}&platform=pc"
+                game_achievements = await self._http_client.get_json_from_request_strict(url)
+                achievements_cache[game_id] = game_achievements["achievements"]
+            return achievements_cache
```

Comment:
I my previous comment I was suggesting to put here content of `update_achievements_cache` method.
I though that this content may be common for all games achievement (apparently I was wrong, as it is static list called once per game_id). 

In this case prepare_achievement_context looses its main benefit: having one-time check done in context of all achievements.
Nevertheless, it is still may be good place to unpack cache from persistent_cache just to increase readability (avoid another Plugin class instance variable by using context instead).

------

I've read once again the whole code and I think, that we can make it much simpler. Looks like Galaxy API requires name OR id of the achievement - see https://github.com/gogcom/galaxy-integrations-python-api/blob/65f4334c032138d88004c418020437a6b571137f/src/galaxy/api/types.py#L129
So if they already prepared achievement definitions on their side, maybe it would be enough to pass only id and avoid caching all_achievements at all?